ArduCopter/commands.cpp

```diff
@@ -107,7 +107,7 @@ bool Copter::far_from_EKF_origin(const Location& loc)
 {
     // check distance to EKF origin
     Location ekf_origin;
-    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M)) {
+    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M) && (abs(ekf_origin.alt - loc.alt) > EKF_ORIGIN_MAX_DIST_M)) {
```

Comment:
I think this "&&" should be "||" because we want it to know if either the horizontal or vertical distance is too far.

Reply:
Got it, Sir